torch/testing/_internal/common_utils.py

```diff
@@ -2853,3 +2853,21 @@ def wrapper(*args, **kwargs):
 def dtype_name(dtype):
     """ Returns the pretty name of the dtype (e.g. torch.int64 -> int64). """
     return str(dtype).split('.')[1]
+
+
+def set_single_threaded_if_parallel_tbb(fn):
+    """Set test to be single threaded for parallel tbb.
+
+    See https://github.com/pytorch/pytorch/issues/64571#issuecomment-914691883
+    """
+    if not IS_TBB:
+        return fn
+
+    @wraps(fn)
+    def wrap_fn(self, *args, **kwargs):
+        num_threads = torch.get_num_threads()
+        torch.set_num_threads(1)
+        output = fn(self, *args, **kwargs)
+        torch.set_num_threads(num_threads)
```

Comment:
should the logic for setting back to the original number of threads be in a finally clause (or context manager) to ensure it gets called even if an exception occurs?